### PR TITLE
[CP-1111] feat: upgrade to php 8.1

### PR DESCRIPTION
**Pull request by Felynx3**
This PR updates PHP version to 8.1 with some extras descripted bellow.

### Changes 
- upgrade php version in composer.json
- upgrade all dependencies
- add tags/branches/php-version relations information table in Readme
- add ci workflow
- add static analysis

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,46 +1,133 @@
...
+        ],
+        "setup": [
+            "@composer install",
+            "@check"
+        ],
+        "check": [
+            "@lint:check",
+            "@security-check",
+            "@test:unit:check",
+            "@test:mutation",
```
Comment (GuillermoFarias) [inline composer.json]: this project use mutations test 👀 ?

**Review by GuillermoFarias — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Felynx3) [reply composer.json]: Oooh nice catch, thank you, I'll fix it

Comment (Felynx3) [reply composer.json]: Fixed

**Review by ibrito182 — DISMISSED** [review]

File: .github/workflows/continuous-integration.yml  [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        with:
+          path: vendor
+          key: ${{ runner.os }}-vendor
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --no-interaction
+
+      - name: Run security check
+        run: composer security-check
+  test-unit-coverage:
+    name: Unit tests, coverage and mutation
```
Comment (GuillermoFarias) [inline .github/workflows/continuous-integration.yml ]: :trollface: se le quedó una mención a  `mutation`

**Review by GuillermoFarias — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous-integration.yml  (hunk 2 above)
Comment (Felynx3) [reply .github/workflows/continuous-integration.yml ]: hahhaa fixed

**Review by GuillermoFarias — APPROVED** [review]

**Review by tamara-dietz — APPROVED** [review]

**Review by rgdevment — APPROVED** [review]

**Pull request merged**